internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1352,6 +1352,11 @@ func (r *BMACReconciler) validateWorkerForDay2(log logrus.FieldLogger, agent *ai
 }
 
 func (r *BMACReconciler) getClusterDeploymentAndCheckIfInstalled(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent) (*hivev1.ClusterDeployment, bool, error) {
+	if agent.Spec.ClusterDeploymentName == nil {
+		log.Debugf("Agent %s/%s is not bind yet", agent.Namespace, agent.Name)
+		return nil, false, nil
```

Comment:
Not sure that it's enough, for example https://github.com/openshift/assisted-service/blob/364ec237c5cc5e564e0e3272155de8d56b8e90bc/internal/controller/controllers/bmh_agent_controller.go#L686  will use `cd.Name` and it will result in a panic
there are probably other places as well.